lsudt/lsudt.py

```diff
@@ -474,10 +496,86 @@ def init_argparse() -> argparse.ArgumentParser:
         dest="id_path",
         help="limit output to devices with a parent starting with given idpath",
     )
+    parser.add_argument(
+        "--extract-env",
+        "-x",
+        action="store_true",
+        dest="extract_env",
+        help="print name=value pairs to stdout"
+    )
 
     return parser
 
 
+def generate_and_print_env_strings(label: str):
+    """
+    prints the name=value pairs to stdout
+    """
+
+    strings = ""
+    for key in list(env_pairs.keys()):
+        counter = 0
+        for dev in env_pairs[key]:
+            strings += (f" {label}_{key}{counter}={dev} ")
```

Comment:
I think you can drop the space after the first quote mark.

Reply:
The output of an SD card looks like this:

    Port 3: {'SDMux': 'SDMUX'} (424:2640 / 1-5.3)
        Port 1: {'Disk': 'DISK'} (424:4050 / 1-5.3.1)
           /dev/sdc
           /dev/sdc1
           /dev/sdc2
           /dev/bsg/7:0:0:0
           /dev/sg1

Which becomes this with the -x option:

RPI4_DISK0=/dev/sdc2  RPI4_DISK1=/dev/sdc  RPI4_DISK2=/dev/bsg/7:0:0:0  RPI4_DISK3=/dev/sg1  RPI4_DISK4=/dev/sdc1

Unfortunately it's unordered, so there is no way to reliably find an env for say the second disk partition, you'd think sdc2 would come after sdc1 etc. I think we need to identify a way to consistently order this list, we could do that alphabetically, but you'd end up with:

RPI4_DISK0=/dev/bsg/7:0:0:0
RPI4_DISK1=/dev/sdc
RPI4_DISK2=/dev/sdc1
RPI4_DISK3=/dev/sdc2
RPI4_DISK4=/dev/sg1

Can we guarantee that there will always be a bsg node, and will sg1 always come after the actually disk? I've noticed the bsg and sg1 nodes are character devices, so I wonder if we can specify as part of the env label that we are only interested in block devices, and then we can list them in alphabetical order?

E.g. 
segments:
  -
    identifier: rpi4
    label: Hub used for RPI4
    ports:
     - port: 1
       label: UART
       env: UART
     - port: 2
       label: USB Relay for Power
       env: POWER
     - port: 3
       label: SDMux
       env: SDMUX
     - port: 3.1
       label: Disk
       env: DISK,b

Notice the ,b at the end of DISK? Or perhaps there is another way.